### PR TITLE
feature: add directory argument

### PR DESCRIPTION
**Pull request by hazcod**
Makes it possible to run the action in a subdirectory.

### CODE REVIEW

**Comment by hazcod** [comment]
CC @mattn @haya14busa

File: README.md [hunk 1]
```diff
@@ -87,6 +90,7 @@ jobs:
           # Can pass --config flag to change golangci-lint behavior and target
           # directory.
           golangci_lint_flags: "--config=.github/.golangci.yml ./testdata"
+          directory: subdirectory/
```
Comment (haya14busa) [inline README.md]: Can you specify sub dir like this with golangci_lint_flags?

Comment (hazcod) [reply README.md]: Does not seem like it: https://github.com/ironPeakServices/iron-redis/pull/15/checks?check_run_id=331138356

**Comment by aslafy-z** [comment]
Related issues:
https://github.com/golangci/golangci-lint/issues/828
https://github.com/golangci/golangci-lint/issues/855
https://github.com/golangci/golangci-lint/issues/865
https://github.com/golangci/golangci-lint/issues/870

Is there any way we could add this feature without breaking the `github-[pr-]check` reporters? We could maybe pass down the sub-path to reviewdog and let him add it to the matched paths?

**Comment by aslafy-z** [comment]
Here is the workaround I use: https://github.com/aslafy-z/action-golangci-lint/blob/master/entrypoint.sh#L9

**Comment by hazcod** [comment]
ping @haya14busa

**Comment by haya14busa** [comment]
>  github-pr-check doesn't work well with subdirectory run.

As I said, github-pr-check won't work with subdir, so I cannot merge it at this point.

bwt, is it really reproducible with latest reviewdog/golangci-lint binary?

I cloned ironPeakServices/iron-redis and ran following command. It succeeded.
```
$  golangci-lint run --enable-all ./...   
```

**Comment by hazcod** [comment]
@haya14busa  seems not: https://github.com/ironPeakServices/iron-redis/pull/15/checks?check_run_id=377848053
`level=error msg="Running error: context loading failed: package github.com/go-redis/redis: no go files to analyze"`

**Comment by haya14busa** [comment]
Can you try v1.1.6 (or v1)?

**Comment by hazcod** [comment]
Nope :/ https://github.com/ironPeakServices/iron-redis/pull/15/checks?check_run_id=377865443

**Comment by evanfuller** [comment]
any update on whether this will merge?

**Comment by samsondav** [comment]
We are also seeing this problem.

File: action.yml [hunk 2]
```diff
@@ -17,9 +17,18 @@ inputs:
   reporter:
     description: 'Reporter of reviewdog command [github-pr-check,github-pr-review,github-check].'
     default: 'github-pr-check'
+  directory:
+    description: 'The subdirectory to execute the action in.'
+    default: ''
 runs:
   using: 'docker'
   image: 'Dockerfile'
+  args:
```
Comment (haya14busa) [inline action.yml:26]: Remove args.

File: action.yml [hunk 3]
```diff
@@ -17,9 +17,18 @@ inputs:
   reporter:
     description: 'Reporter of reviewdog command [github-pr-check,github-pr-review,github-check].'
     default: 'github-pr-check'
+  directory:
```
Comment (haya14busa) [inline action.yml:20]: Can you rename it to `workdir`?  It's consistent with https://github.com/reviewdog/action-eslint/pull/19 and I prefer workdir to directory.

**Comment by review-dog** [comment]
Hi, @hazcod! We merged your PR to reviewdog! 🐶
Thank you for your contribution! ✨

**We just invited you to join the @reviewdog organization on GitHub.**
Accept the invite by visiting https://github.com/orgs/reviewdog/invitation.
By joining the team, you'll be a part of reviewdog community and can help the maintainance of reviewdog.

Thanks again!

**Pull request merged**